src/main/java/com/iota/iri/service/milestone/impl/LatestMilestoneTrackerImpl.java

```diff
@@ -209,8 +208,13 @@ public MilestoneValidity analyzeMilestoneCandidate(TransactionViewModel transact
                     transaction.getCurrentIndex() == 0) {
 
                 int milestoneIndex = milestoneService.getMilestoneIndex(transaction);
+                if (milestoneIndex <= snapshotProvider.getInitialSnapshot().getIndex()) {
+                    return INVALID;
+                }
```

Comment:
I think Alon's beef was that the `validateMilestone` depended on the milestone Index. On which he was correct, we don't want other potential callers of this method to be blocked by the index.

 He didn't say anything about your `IRRELEVANT` enum. I think that returning an INVALID on a vaild milestone will be very misleading to anyone in the future (it is at least misleading for me). I would bring back this enum value.

Reply:
Actually I have a better idea... Instead of adding another enum to validity (IRRELEVANT) doesn't make much sense, `analyzeMilestoneCandidate` can return a boolean: `true` if successfully analyzed, `false` otherwise.

If the `milestoneIndex` is below the snapshot then return `true` and add a little comment: "has already been analyzed if it is below the snapshot" (feel free to change the wording).

I think this makes the most sense.

Reply:
I agree w/ @GalRogozinski that returning `INVALID` may not be the best answer. other than that lgtm.

Reply:
i changed it accordingly